query/common.d.ts

```diff
@@ -0,0 +1,278 @@
+/* Query Interfaces - Common */
+/* https://rdf.js.org/query-spec/ */
+
+import { EventEmitter } from "events";
+import * as RDF from '../data-model';
+
+/**
+ * Helper union type for quad term names.
+ */
+export type QuadTermName = 'subject' | 'predicate' | 'object' | 'graph';
+
+/**
+ * Custom typings for the RDF/JS ResultStream interface as the current
+ * typings restrict the generic param Q to extensions of "BaseQuad",
+ * meaning it cannot be used for Bindings.
+ */
+export interface ResultStream<Q> extends EventEmitter {
+  read(): Q | null;
```

Comment:
Reference: https://github.com/rdfjs/types/blob/b04a4158c7dabbcd487661255afbe7c9eac5f25f/stream.d.ts#L22

Reply:
We could have a `Stream<Q extends BaseQuad | Bindings>`, a `QuadStream<Q extends BaseQuad = Quad> extends Stream<Q>` and a `BindingsStream extends Stream<Bindings>`

Reply:
Yes, that's possible indeed. But the problem with that is that this will imply a breaking change in the typings, which might not be pleasant for downstream users.
Perhaps we can add a TODO to merge them from the moment we require a breaking change?

Reply:
True! Yes, I think a TODO would be a good idea.